packages/mongo-nav/src/MongoNav.tsx

```diff
@@ -157,32 +167,44 @@ const navContainerStyle = css`
  * @param props.activePlatform Determines which platform is active
  * @param props.alertPollingInterval Defines interval for alert polling
  */
-function MongoNav({
-  activeProduct,
-  activeNav,
-  onOrganizationChange,
-  onProjectChange,
-  activeOrgId,
-  activeProjectId,
-  className,
-  environment = Environment.Commercial,
-  activePlatform = Platform.Cloud,
-  mode = Mode.Production,
-  loadData = true,
-  showProjectNav = true,
-  onError = () => {},
-  onSuccess = () => {},
-  onElementClick = () => {},
-  onPrem = { mfa: false, enabled: false, version: '' },
-  alertPollingInterval = 600e3, // 10 minutes
-  admin: adminProp,
-  hosts: hostsProp,
-  urls: urlsProp,
-  constructOrganizationURL: constructOrganizationURLProp,
-  constructProjectURL: constructProjectURLProp,
-  dataFixtures,
-  ...rest
-}: MongoNavInterface) {
+function MongoNav(
+  {
+    activeProduct,
+    activeNav,
+    onOrganizationChange,
+    onProjectChange,
+    activeOrgId,
+    activeProjectId,
+    className,
+    environment = Environment.Commercial,
+    activePlatform = Platform.Cloud,
+    mode = Mode.Production,
+    loadData = true,
+    showProjectNav = true,
+    onError = () => {},
+    onSuccess = () => {},
+    onElementClick = () => {},
+    onPrem = { mfa: false, enabled: false, version: '' },
+    alertPollingInterval = 600e3, // 10 minutes
+    admin: adminProp,
+    hosts: hostsProp,
+    urls: urlsProp,
+    constructOrganizationURL: constructOrganizationURLProp,
+    constructProjectURL: constructProjectURLProp,
+    dataFixtures,
+    ...rest
+  }: MongoNavInterface,
+  ref: React.Ref<any>,
+) {
+  const [reload, setReload] = useState(false);
+
+  useImperativeHandle(ref, () => ({
+    reloadData: () => {
+      console.log('inside the exposed property');
+      setReload(curr => !curr);
```

Comment:
Good call 👀 @lazytype

Reply:
could it be renamed to `reloadToggle` or something like that to indicate that whether it's true/false doesn't matter? alternatively do something like `forceUpdateCounter`, e.g. `forceReloadCounter`

Reply:
what do think should be renamed `reloadData` or `setReload`

Reply:
Also am having trouble with my attempts at testing this... not sure if you have any ideas

Reply:
> what do think should be renamed reloadData or setReload

sorry, I meant `setReload` -> `setReloadToggle`/`setForceReloadCounter`.

> Also am having trouble with my attempts at testing this... not sure if you have any ideas

what sort of issue are you encountering? I'd expect the high-level structure of the test to look something like:
```
// mock fetch data
// render mongo nav
// wait for screen to display the data
// mock fetch with different data
// call reloadData
// wait for screen to display the new data
```

```

Reply:
Will rename right now and try this... i was trying something like: 

``` 
 describe('able to access "reload" on a ref passed to MongoNav', () => {
    function MongoNavWrapper() {
      const mongoNavRef = React.useRef();

      setTimeout(() => {
// Below throwing error, not logging
         console.log(Object.keys(mongoNavRef.current));
        mongoNavRef.current.reloadData();
      }, 5000);

     

      return (
        <>
          <MongoNav mode="dev" ref={mongoNavRef} />
          <div>{mongoNavRef.current}</div>
        </>
      );
    }

    beforeEach(async () => {
      await render(<MongoNavWrapper />);
    });

    test('test', () => {
      // console.log(getByTestId('user-menu-trigger'));
    });
  });```

Reply:
if you need to get access to the `reloadData` function in the test, maybe something like this will work:
```
let ref;
function MongoNavWrapper() {
  ref = React.useRef();
  ...
}

...

ref.current.reloadData(); // maybe this needs `act`?
```

Reply:
yes got it!